### PR TITLE
feat(List): selectedIndex && a11y enhancements 

### PR DESCRIPTION
**Pull request by mgr34**
This achieves the same functionality as PR #589. I have watered my old branch and made the necessary changes. This would implement #554  (which should be reopened)

### CODE REVIEW

File: src/list/readme-variants.tsx [hunk 1]
```diff
@@ -146,26 +146,46 @@ export default function Readme() {
...
         {function Example() {
-          const [checked, setChecked] = React.useState<any>({
-            Cookies: false,
-            Pizza: false,
-            Icecream: false
-          });
+          const shoppingList = ['Cookies','Pizza','Icecream']
+          const [selectedIndex, setSelectedIndex] = React.useState<number[]>([])
+
+          //@ts-ignore
```
Comment (EmiBemi) [inline src/list/readme-variants.tsx:172]: is ts-ignore need here? What's the typescript error? (I can check out the branch and have a look tomorrow)

File: src/list/list.tsx [hunk 2]
```diff
@@ -28,6 +29,10 @@ export interface ListProps {
   wrapFocus?: boolean;
   /** Sets the lists vertical orientation. Defaults to true */
   vertical?: boolean;
+  /** sets the selectedIdex for singleSelection, radiogroup, or checkboxlist variants. Only supply number[] to checkboxlists */
```
Comment (EmiBemi) [inline src/list/list.tsx:32]:
```suggestion
  /** Sets the selectedIndex for singleSelection, radiogroup, or checkboxlist variants. Only supply number[] to checkboxlists */
```

File: src/list/list.tsx [hunk 3]
```diff
@@ -57,12 +62,39 @@ export const List = createComponent<ListProps>(function List(props, ref) {
...
   } = props;
-  const { rootEl } = useListFoundation({
-    ...props,
-    wrapFocus
-  });
+  const { rootEl, listItemClasses, setEnabled, role} = useListFoundation(props);
+
+  
+  const getListItemRole = (): {role?: string} => {
+    // TODO(mgr34): menuitems with checkboxs or radios should be
```
Comment (EmiBemi) [inline src/list/list.tsx:74]: Would be good to create an issue for this so we won't forget

**Comment by b-d-m-p** [comment]
@mgr34 we are sorry for the delay. Are these changes still relevant? Can you update the branch with the latest commits? We promise to act quick after that.

**Comment by mgr34** [comment]
Hi @b-d-m-p, 
 Pretty busy week ahead of me, but I'll try to circle around to this in the next couple of weeks. I believe it to be still relevant.

**Comment by b-d-m-p** [comment]
@mgr34 Sounds good! We are excited to get a11y improvements because it's newly required in EU law, so it will be very helpful. Looking forward to hearing from you. Cheers.

**Comment by b-d-m-p** [comment]
@mgr34 Just checking in. We might have a go at if you're busy at the moment. Let us know. Cheers.

**Comment by jensborch** [comment]
@mgr34 You changes has finally been merged into master in another PR, see https://github.com/rmwc/rmwc/commit/2001413cb3945c440324e1684f6414a4db1ca17f. I will therefor close this. Thanks for helping out :-)

**Pull request closed without merging**